### ISSUE TITLE
QueryStringQueryBuilderTests testToQueryFuzzyQueryAutoFuziness falilure 

### ISSUE THREAD

**Issue by pgomulka** [elastic/elasticsearch#43318]
Failed only once on 7.x, but is reproducible
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(_source),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:QueryStringQueryBuilderTests),sort:!(process.time-start,desc))

https://scans.gradle.com/s/fajagcab7d2ko/tests/nfxodyych4uas-dc5bjgeyodbyk?openStackTraces=WzEsMiwwXQ

reproduces locally:
```
REPRODUCE WITH: ./gradlew :server:test --tests "org.elasticsearch.index.query.QueryStringQueryBuilderTests.testToQueryFuzzyQueryAutoFuziness" -Dtests.seed=98112E2CF02F0BD9 -Dtests.security.manager=true -Dtests.locale=ar-YE -Dtests.timezone=Chile/EasterIsland -Dcompiler.java=12 -Druntime.java=8
```

stacktrace
```
org.elasticsearch.index.query.QueryShardException: Failed to parse query [OR~]Close stacktrace
at __randomizedtesting.SeedInfo.seed([98112E2CF02F0BD9:EC37F5B85574AAB2]:0)
at org.elasticsearch.index.query.QueryStringQueryBuilder.doToQuery(QueryStringQueryBuilder.java:914)
at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:99)
at org.elasticsearch.index.query.QueryStringQueryBuilderTests.testToQueryFuzzyQueryAutoFuziness(QueryStringQueryBuilderTests.java:814)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.lucene.queryparser.classic.ParseException: 
Cannot parse 'OR~': Encountered " <OR> "OR "" at line 1, column 0.
Was expecting one of:
    <NOT> ...
    "+" ...
    "-" ...
    <BAREOPER> ...
    "(" ...
    "*" ...
    <QUOTED> ...
    <TERM> ...
    <PREFIXTERM> ...
    <WILDTERM> ...
    <REGEXPTERM> ...
    "[" ...
    "{" ...
    <NUMBER> ...
    <TERM> ...
    ```

For other seeds it passes
```
./gradlew :server:test --tests "org.elasticsearch.index.query.QueryStringQueryBuilderTests.testToQueryFuzzyQueryAutoFuziness" \
  -Dtests.seed=98112E2CF02F0BD8 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-YE \
  -Dtests.timezone=Chile/EasterIsland \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

**Comment by pgomulka** [comment]
could it be failing because it generated a query with a logical operator "OR", this will also fail for "AND", "NOT"

**Comment by jimczi** [comment]
> could it be failing because it generated a query with a logical operator "OR", this will also fail for "AND", "NOT"

Yep that's the issue. I don't think we need to produce a random string here so we could change the test to always use the same letter in the sequence (`aa`, `aaa`) to avoid any reserved keywords.

**Comment by droberts195** [comment]
This can still fail on the 6.8 branch.  For example:

```
./gradlew ':server:unitTest' -Dtests.seed=9146B2ABAEC4241E -Dtests.class=org.elasticsearch.index.query.QueryStringQueryBuilderTests -Dtests.method="testToQueryFuzzyQueryAutoFuziness" -Dtests.security.manager=true -Dtests.locale=fi-FI -Dtests.timezone=America/Rosario -Dcompiler.java=12 -Druntime.java=11
```

Is it worth backporting the fix to 6.8?

**Comment by cbuescher** [comment]
@droberts195 makes sense, even though this fails really rarely, but backport is easy in this case and saves someone in future test triage some trouble.  I opened #51531 to get a backport test run.
